internal/dag/conditions_test.go

```diff
@@ -270,11 +271,37 @@ func TestConditionsValid(t *testing.T) {
 			}},
 			want: false,
 		},
+		"invalid prefix condition": {
+			conditions: []projcontour.Condition{{
+				Prefix: "api",
+			}},
+			want: false,
+		},
+		"invalid prefix condition with headers": {
+			conditions: []projcontour.Condition{{
+				Prefix: "api",
+				Header: &projcontour.HeaderCondition{
+					Name:     "x-header",
+					Contains: "abc",
+				},
+			}},
+			want: false,
+		},
 	}
 
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
-			got := pathConditionsValid(tc.conditions)
+			// TODO(youngnick) This feels dirty but I don't know a better solution.
+			// We're not testing the returned info from the ObjectStatusWriter here,
+			// that's done at the status_test.go layer.
+			swblank := &ObjectStatusWriter{}
+			sw, _ := swblank.WithObject(&projcontour.HTTPProxy{
```

Comment:
yeah this is a bit unfortunate. Perhaps in the future status writer should become an interface then we can supply a dummy.

Reply:
Ah, that makes sense. Consider my mental note made, but I'll put a TODO there and make an issue.